src/redux/sagas/motions/expenditures/cancelExpenditureMotion.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  ClientType,
+  Id,
+  getPermissionProofs,
+  ColonyRole,
+} from '@colony/colony-js';
+import { call, put, takeEvery } from 'redux-saga/effects';
+
+import { ADDRESS_ZERO, APP_URL } from '~constants/index.ts';
+import { type Action, ActionTypes } from '~redux/index.ts';
+import {
+  createGroupTransaction,
+  createTransactionChannels,
+  waitForTxResult,
+} from '~redux/sagas/transactions/index.ts';
+import {
+  getColonyManager,
+  initiateTransaction,
+  putError,
+  takeFrom,
+} from '~redux/sagas/utils/index.ts';
+
+function* cancelExpenditureMotion({
+  meta,
+  meta: { setTxHash },
+  payload: { colony, votingReputationAddress, expenditure, motionDomainId },
+}: Action<ActionTypes.MOTION_EXPENDITURE_CANCEL>) {
+  const { createMotion } = yield call(createTransactionChannels, meta.id, [
+    'createMotion',
+  ]);
+
+  try {
+    if (
+      !colony ||
+      !expenditure ||
+      !votingReputationAddress ||
+      !motionDomainId
+    ) {
+      throw new Error('Invalid payload');
+    }
```

Comment:
We have this pattern in most sagas. I guess it would be a good idea to actually convert it across all of them to some sort of util.

But yeah, that's just some more for the maintenance pile for which we don't really have time right now.